### ISSUE TITLE
Snapshot fold should be from child to self

### ISSUE THREAD

**Issue by yasker** [rancher/longhorn#108]
Rather than self to parent.

For example,

```
rancher@rancher:~$ ./longhorn snapshot
ID
71c143d2-5145-460a-9e39-176b9d8054bc
b88ffa4b-ea65-4847-90b1-32fb9cf7aba4
ac2502ee-0c56-453c-adfb-1f61a34604c5
rancher@rancher:~$ ./longhorn ls
ADDRESS              MODE CHAIN
tcp://localhost:9502 RW   [volume-head-003.img volume-snap-71c143d2-5145-460a-9e39-176b9d8054bc.img volume-snap-b88ffa4b-ea65-4847-90b1-32fb9cf7aba4.img volume-snap-ac2502ee-0c56-453c-adfb-1f61a34604c5.img]
rancher@rancher:~$ ./longhorn snapshot rm b88ffa4b-ea65-4847-90b1-32fb9cf7aba4
INFO[0000] Coalescing b88ffa4b-ea65-4847-90b1-32fb9cf7aba4 on tcp://localhost:9502
deleted b88ffa4b-ea65-4847-90b1-32fb9cf7aba4
rancher@rancher:~$ ./longhorn ls
ADDRESS              MODE CHAIN
tcp://localhost:9502 RW   [volume-head-003.img volume-snap-71c143d2-5145-460a-9e39-176b9d8054bc.img volume-snap-ac2502ee-0c56-453c-adfb-1f61a34604c5.img]
```

`b88ffa4b-ea65-4847-90b1-32fb9cf7aba4` content should be in `71c143d2-5145-460a-9e39-176b9d8054bc` rather than `ac2502ee-0c56-453c-adfb-1f61a34604c5`

Can be tested after `snapshot revert` feature merged.

**Comment by yasker** [comment]
Fixed by #143

**Comment by sangeethah** [comment]
Tested with longhorn version - cjellick/longhorn:dev-8698fe1 and latest cattle build form jimeng's branch.

Pre Req for each of the cases:
Deploy Vms with ROOT and DATA disk using longhorn drivers.
Take 3 snapshots of ROOT/DATA disk.

scenario1:
Delete 1 st snapshot . Stop VM.Revert to 2nd snapshot. Start VM.
scenario2:
Delete 2nd snapshot .Stop VM.Revert to 2nd snapshot. Start VM.
scenario3:
Delete 2nd snapshot .Stop VM.Revert to 3rd snapshot. Start VM.

In all the the above cases , contents of the disk after revert is as expected ( matches the snapshot contents it reverts to).
